attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -138,6 +165,23 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
 
         let challenge = resp.json::<Challenge>().await?;
         debug!("get challenge: {challenge:#?}");
+
+        let extra_params = challenge.extra_params;
+
+        let algorithm = match extra_params.get(SELECTED_HASH_ALGORITHM_JSON_KEY) {
+            Some(selected_hash_algorithm) => {
+                // Note the blank string which will be handled as an error when parsed.
+                let name = selected_hash_algorithm
+                    .as_str()
+                    .unwrap_or("")
```

Comment:
nit: i'd use `.ok_or(Error::NotAString(selected_hash_algorithm))?` or something, otherwise we might swallow a malformed extra_params body